scripts/cURL/Kalkun_API.php

```diff
@@ -121,12 +139,49 @@ function send_sms()
 			'sms_mode' => urlencode($this->sms_mode),
 			'sms_loop' => urlencode('1'),
 			'validity' => urlencode('-1'),
-			'unicode' => urlencode($this->coding),
-			'message' => urlencode($this->message)
+			'message' => urlencode($this->message),
+			'csrf_kalkun_tkn' => $csrf_hash,
 		);
 		$sms_field = $this->urlify($sms);
 		curl_setopt($ch, CURLOPT_POSTFIELDS, $sms_field);
 		$output = curl_exec($ch);
+
+		return curl_getinfo($ch, CURLINFO_HTTP_CODE);
+	}
+
+	function get_csrf_hash()
+	{
+		$ch = $this->curl_id;
+		curl_setopt($ch, CURLOPT_URL, $this->csrf_hash_url);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, TRUE);
+		curl_setopt($ch, CURLOPT_POST, FALSE);
+		curl_setopt($ch, CURLOPT_COOKIEJAR, $this->session_file);
+		curl_setopt($ch, CURLOPT_COOKIEFILE, $this->session_file);
+		$output = curl_exec($ch);
+		return json_decode($output);
+	}
+
+	function get_csrf_hash_from_login_form()
```

Comment:
Any reason to have 2 function for getting csrf token? Wouldn't just get_csrf_hash() suffice?

Reply:
No. This one needs to be logged in